pkg/apiserver/resthandler.go

```diff
@@ -347,6 +347,21 @@ func DeleteResource(r RESTDeleter, ctxFn ContextFunc, namer ScopeNamer, codec ru
 			ctx = api.WithNamespace(ctx, namespace)
 		}
 
+		options := &api.DeleteOptions{}
+		if checkBody {
+			body, err := readBody(req.Request)
+			if err != nil {
+				errorJSON(err, codec, w)
+				return
+			}
+			if len(body) > 0 {
```

Comment:
It might make sense to create options if none were provided and then run them through the defaulting pass. Otherwise, code all over will presumably need to reason about default behavior in the absence of options.